File: app/src/main/java/com/x8bit/bitwarden/WebAuthCallbackActivity.kt
package com.x8bit.bitwarden

import android.content.Intent
import android.os.Bundle
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import com.x8bit.bitwarden.data.platform.annotation.OmitFromCoverage
import dagger.hilt.android.AndroidEntryPoint

/**
 * An activity to receive callbacks from Custom Chrome tabs or other web-auth related flows such
 * the current state of the task holding the [MainActivity] can remain undisturbed.
 */
@OmitFromCoverage
@AndroidEntryPoint
class WebAuthCallbackActivity : AppCompatActivity() {

    private val webAuthCallbackViewModel: WebAuthCallbackViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        webAuthCallbackViewModel.trySendAction(
            WebAuthCallbackAction.IntentReceive(intent = intent),
        )

        val intent = Intent(this, MainActivity::class.java)
            .apply {
                addFlags(
                    Intent.FLAG_ACTIVITY_CLEAR_TOP or
                        Intent.FLAG_ACTIVITY_SINGLE_TOP,
                )
            }
        startActivity(intent)
        finish()
    }
}

File: src/App/Platforms/Android/WebAuthCallbackActivity.cs
using Android.App;
using Android.Content.PM;
using Android.OS;
using Android.Runtime;
using Bit.App.Droid.Utilities;

namespace Bit.Droid
{
    [Register("com.x8bit.bitwarden.WebAuthCallbackActivity")]
    public class WebAuthCallbackActivity : WebAuthenticatorCallbackActivity
    {
        protected override void OnCreate(Bundle savedInstanceState)
        {
            Intent?.Validate();
            base.OnCreate(savedInstanceState);
        }
    }
}
